src/cdk/a11y/high-contrast-mode/high-contrast-mode-detector.ts

```diff
@@ -45,17 +47,28 @@ export class HighContrastModeDetector implements OnDestroy {
    * Figuring out the high contrast mode and adding the body classes can cause
    * some expensive layouts. This flag is used to ensure that we only do it once.
    */
-  private _hasCheckedHighContrastMode: boolean;
+  private _hasCheckedHighContrastMode: boolean = false;
+  private _forcedColorsActive: boolean = false;
   private _document: Document;
   private _breakpointSubscription: Subscription;
 
-  constructor(private _platform: Platform, @Inject(DOCUMENT) document: any) {
-    this._document = document;
+  constructor(private _platform: Platform, @Inject(DOCUMENT) doc: any) {
+    this._document = doc;
+
+    // Mark if there's a HighContrastMode detector active on the body so if multiple
```

Comment:
I suspect that the `HighContrastModeDetector` isn't the only place where we have issues like this since we generally assume only one copy of Angular will be on the page. Should we tackle this across the entire library?